test_suite/tests/derive.rs

```diff
@@ -158,11 +176,11 @@ fn c_like_enum_derive_with_scale_index_set() {
 
     let ty = Type::builder().path(Path::new("E", "derive")).variant(
         Variants::fieldless()
-            .variant("A", 0)
-            .variant("B", 10)
-            .variant("C", 13)
-            .variant("D", 3)
-            .variant("E", 14),
+            .variant("A", 0, &[])
+            .variant("B", 10, &[])
+            .variant("C", 13, &[])
+            .variant("D", 3, &[])
+            .variant("E", 14, &[]),
```

Comment:
I'd actually prefer an API where I could do:
```rust
.variant("A", 0).docs(&[])
```
instead, where the `docs(..)` is not mandatory. so if you leave it away docs are naturally empty.

Reply:
I agree such an API would be nicer. It would require a fair amount of refactoring to achieve though. For other cases in the past we've said that it's ok-ish to have an ugly API given it is rare for humans to use it: most types are derived.
That said, if it's possible, I'd prefer it as well. We'd likely end up with having to call `finalize()` explicitly (which imo is fine), like `.variant("A", 0).docs(&[]).finalize()` (or something similar).